File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/tei_data/DashboardProgramViewHolder.java
package org.dhis2.usescases.teiDashboard.dashboardfragments.tei_data;

import androidx.recyclerview.widget.RecyclerView;

import org.dhis2.BR;
import org.dhis2.Bindings.Bindings;
import org.dhis2.databinding.ItemDashboardProgramBinding;
import org.dhis2.usescases.teiDashboard.DashboardProgramModel;
import org.hisp.dhis.android.core.enrollment.Enrollment;
import org.hisp.dhis.android.core.program.Program;

/**
 * QUADRAM. Created by ppajuelo on 29/11/2017.
 */

class DashboardProgramViewHolder extends RecyclerView.ViewHolder {
    private ItemDashboardProgramBinding binding;

    DashboardProgramViewHolder(ItemDashboardProgramBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(TEIDataContracts.Presenter presenter, DashboardProgramModel dashboardProgramModel, int position) {
        Program program = dashboardProgramModel.getEnrollmentPrograms().get(position);
        Enrollment enrollment = dashboardProgramModel.getEnrollmentForProgram(program.uid());
        binding.setVariable(BR.presenter, presenter);
        binding.setVariable(BR.program, program);

        Bindings.setObjectStyle(binding.programImage, binding.programImage, dashboardProgramModel.getObjectStyleForProgram(program.uid()));

        if (enrollment != null)
            binding.setVariable(BR.enrollment, enrollment);
        binding.executePendingBindings();

        itemView.setOnClickListener(v -> presenter.setProgram(dashboardProgramModel.getEnrollmentPrograms().get(position), enrollment.uid()));
    }
}

File: app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/teidata/DashboardProgramViewHolder.kt
package org.dhis2.usescases.teiDashboard.dashboardfragments.teidata

import androidx.recyclerview.widget.RecyclerView
import org.dhis2.BR
import org.dhis2.databinding.ItemDashboardProgramBinding
import org.dhis2.usescases.teiDashboard.DashboardTEIModel
import org.hisp.dhis.android.core.program.Program

class DashboardProgramViewHolder(
    private val binding: ItemDashboardProgramBinding,
) : RecyclerView.ViewHolder(binding.root) {

    fun bind(
        presenter: TEIDataPresenter,
        dashboardProgramModel: DashboardTEIModel,
        position: Int,
    ) {
        dashboardProgramModel.getProgramsWithActiveEnrollment()?.get(position)?.let { program: Program ->
            val enrollment = dashboardProgramModel.getEnrollmentForProgram(program.uid())
            binding.setVariable(BR.presenter, presenter)
            binding.setVariable(BR.program, program)
            binding.metadataIconData = dashboardProgramModel.getIconForProgram(program.uid())
            if (enrollment != null) {
                binding.setVariable(BR.enrollment, enrollment)
                binding.setVariable(
                    BR.orgUnit,
                    presenter.getOrgUnitName(enrollment.organisationUnit() ?: ""),
                )
            }
            binding.executePendingBindings()
            itemView.setOnClickListener {
                presenter.setProgram(
                    program,
                    enrollment!!.uid(),
                )
            }
        }
    }
}
